internal/keepers/ocr.go

```diff
@@ -57,7 +57,22 @@ func (k *keepers) Observation(ctx context.Context, rt types.ReportTimestamp, _ t
 	// should be more uniform for all nodes
 	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
 
-	b, err := limitedLengthEncode(keys, maxObservationLength)
+	obs := &ktypes.UpkeepObservation{
+		UpkeepIdentifiers: []ktypes.UpkeepIdentifier{},
+	}
+	// TODO get latest block from head ticker
```

Comment:
We need to be able to emit an `UpkeepObservation` with a block key and an empty list of upkeep identifiers. Right now, the code relies on a non-empty list of upkeep identifiers to get the block key, but in the scenario where we have an empty list of upkeep identifiers, we need to get the block key from *somewhere*. My understanding is that the HeadTicker should be able to provide this?

Reply:
evm registry has a function `latestBlock`, we can just expose that https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evm/registry.go#L134-L135 ?

Reply:
It would be easiest to add a function to the `upkeepService` interface that exposes the latest block key. The latest block key comes to the service in the the following:

```
func (s *onDemandUpkeepService) start() {
	// TODO: if this process panics, restart it
	go s.cacheCleaner.Run(s.cache)
	go func() {
		ch := s.headSubscriber.HeadTicker()
		for {
			select {
			case <-ch:
				// run with new head
				s.processLatestHead(s.ctx)
			case <-s.ctx.Done():
				return
			}
		}
	}()
}
```

Just keep track of the last known block key from the head ticker channel and return that to the OCR process.

Reply:
worth noting here that the channel was forced to have size 1, the behaviour we wanted was to skip blocks while `s.processLatestHead` was running. Not sure if it causes issues, just something to keep in mind

Reply:
thinking more, this might be the behaviour we actually want. We do not send the 'latest' head, but the latest head we are checking the upkeeps on. 
This will keep it consistent with the case when we do actually have upkeeps which are eligible

Reply:
resolved, using blockkey from head ticker